pkg/backend/httpstate/client/ai.go

```diff
@@ -54,8 +66,37 @@ func createSummarizeUpdateRequest(
 	}
 }
 
-// extractSummaryFromResponse parses the Copilot API response and extracts the summary content
-func extractSummaryFromResponse(copilotResp apitype.CopilotSummarizeUpdateResponse) (string, error) {
+// createExplainPreviewRequest creates a new CopilotExplainPreviewRequest with the given content and org ID
+func createExplainPreviewRequest(
+	lines []string,
+	orgID string,
+	maxUpdateOutputLen int,
+) apitype.CopilotExplainPreviewRequest {
+	content := strings.Join(lines, "\n")
```

Comment:
Nit: Any value in accepting lines and then immediately joining them, as opposed to accepting a single block of text and joining in the display? Super minor but keeps the formatting logic in one place.

Reply:
No, I had a slightly misguided assumption about golang copying strings by value and this string can be quite large. Reading up on it this is not the case, strings seem to be immutable like python and so its not actually copied when passing by value.